spec/app/ics-027-interchain-accounts/README.md

```diff
@@ -20,110 +20,126 @@ On Ethereum, there are two types of accounts: externally owned accounts, control
+- The authentication step on the recieving chain (where the interchain account is hosted) fails 
 
-It does not restrict how you can distinguish signers that was not made by the counterparty chain. But the most common way would be to record the account in state when the interchain account is registered and to verify that signers are recorded interchain account.
+#### Known Issues
+##### Ordered Channels 
+In an ordered channel, packet 2 cannot be relayed until packet 1 has been relayed or a valid proof that packet 1 has timed out has been recieved. If multiple distrusting accounts are allowed to use a single ordered channel this creates an attack vector whereby malicious users can block packets from being recieved in a timely manner. An example of this may be a malicious user sending hundreds of packets that are never paid for. Other users trying to use this channel will have to wait for all of these packets to be fully relayed or recieve valid proof of these packets having timed out before their own packets can be relayed. Therefore, multiplexing N interchain accounts on 1 ordered channel is not viable. This specification assumes the following: N interchain accounts will exist on N ordered channels. 
 
-### Data Structures
+##### Unordered Channels 
+An unordered channel lets packets be relayed in any order. A user can send 100 packets and never relay them. Another user submits one packet and can immediately relay just that one without having to relay all other packets. In theory interchain accounts can be implemented using an unordered channel so long as messages that are dependent on one another are packed into a single packet. However, this pattern differs from other standards such as how Tendermint's mempool operates. Rather than deviate from well defined standards for message processing it is recommended to not to use this approach or at a minimum clearly document how messages are processed on the recieving chain. 
 
-Each chain must implement the interfaces as defined in this section in order to support interchain accounts.
+##### Partially Ordered Channels 
+Interchain accounts is a perfect use case for partially ordered channels, whereby the order is based on account sequences. However, this has yet to be implemented in IBC. As of the time of writing this specification the current progress can be tracked [here](https://github.com/cosmos/ibc/issues/550). 
 
-`tryRegisterIBCAccount` method in `IBCAccountModule` interface defines the way to request the creation of an IBC account on the host chain (or counterparty chain that the IBC account lives on). The host chain creates an IBC account using its account creation logic, which may use data within the packet (such as destination port, destination channel, etc) and other additional data for the process. The origin chain can receive the address of the account that was created from the acknowledge packet.
+### Architecture Diagram
+![](https://i.imgur.com/HX1h2u2.png)
```

Comment:
I hosted this on imgur just for now. Where should we keep this asset?

Reply:
https://giphy.com/ :trollface: 

But seriously, can't you commit the png to this repo and link it in the same directory?

Reply:
Yeah, that's the obvious solution. I just wanted to determine a protocol for this. All the images in the other specs are currently broken.

Reply:
Any thoughts on the actual diagram in general?

Reply:
I don't really understand the arrows, why are arrows going to the relayer? The ordering (numbers) makes sense to me.

Reply:
Thanks for the feedback, perhaps we can remove/improve the arrows in some way. 

In this context, we're calling onChanOpenInit and emitting an event, which the relayer picks up and then begins the handshake process on both chains for creating a channel. The arrows are trying to symbolize that the relayer is both listening for events + coordinating the handshake process. I guess this isn't being communicated clearly tho.

Reply:
I see - yes, that's true, maybe labelling the arrows or something would help.

Reply:
I removed this in the end.